src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -168,15 +170,27 @@ private RequestDelegate BuildApplication()
                 // Generate an HTML error page.
                 var hostingEnv = _applicationServices.GetRequiredService<IHostingEnvironment>();
                 var showDetailedErrors = hostingEnv.IsDevelopment() || _options.DetailedErrors;
-                var errorBytes = StartupExceptionPage.GenerateErrorHtml(showDetailedErrors, ex);
 
+                var model = new ErrorPageModel();
+                if (showDetailedErrors)
+                {
+                    var exceptionDetailProvider = new ExceptionDetailsProvider(
+                        hostingEnv.ContentRootFileProvider,
+                        sourceCodeLineCount: 6);
```

Comment:
Nothing magical. It's just the same number we use everywhere. No need to make it configurable.